subsys/net/lib/nrf_cloud/src/nrf_cloud.c

```diff
@@ -529,11 +529,12 @@ void nrf_cloud_run(void)
 			LOG_DBG("Socket error: POLLNVAL");
 			if (nfsm_get_disconnect_requested()) {
 				LOG_DBG("The cloud socket was disconnected by request");
+				evt.status = NRF_CLOUD_DISCONNECT_USER_REQUEST;
```

Comment:
Does this make line 317 of `nrf_cloud_fsm.c` redundant?

Reply:
https://github.com/nrfconnect/sdk-nrf/blob/abfa889337da0c1539389038428098f050fdd5cb/subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c#L317

Reply:
i don't think so, but feel free to prove me wrong.

Reply:
I'm not 100% convinced it does. I'm forgetting how the FSM handles disconnections. I'll toy with it for a few minutes and see what happens, but if I find nothing I'm happy leaving it as-is

Reply:
you'd need to also toy with `CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD` disabled 🙈

Reply:
Oh, yeah, you are totally right. Not only does your note about CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD prove that without further inspection, upon further inspection I understand why you are doing things this way. Thanks for the hint.

LGTM